bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java

```diff
@@ -1184,56 +1184,50 @@ public int shutdown() {
     }
     // internal shutdown method to let shutdown bookie gracefully
     // when encountering exception
-    AtomicBoolean shuttingDown = new AtomicBoolean(false);
-    private final CountDownLatch countDownLatch = new CountDownLatch(1);
+    ReentrantLock lock = new ReentrantLock(true);
     int shutdown(int exitCode) {
         try {
+            lock.lock();
```

Comment:
The lock should be placed before the try.

Reply:
fixed